### ISSUE TITLE
OverloadedRecordDot usage doesn't trigger redundant bracket warning

### ISSUE THREAD

**Issue by ocharles** [ndmitchell/hlint#1471]
The following code triggers a redundant bracket warning:

```haskell
    referenceDesignator = ReferenceDesignator (placementReferenceDesignator)
```

However, the following doesn't:

```haskell
    referenceDesignator = ReferenceDesignator (p.placementReferenceDesignator)
```

This is a bug, as the brackets here are redundant.

**Comment by benbellick** [comment]
@zliu41 I would like to start by trying to take on this issue. I have been familiarizing myself with the codebase, but if I have questions that come up, can I ask them to you? If so, is this the best place to ask?

**Comment by zliu41** [comment]
Yes, feel free to post your questions here.

**Comment by benbellick** [comment]
@zliu41 I have opened a PR. I fixed the problem by treating x.y as an atom. Does that make sense in context? I am honestly not sure what exactly Haskell considers an atom versus not. 

Also, is there a generally good way to familiarize yourself with a Haskell codebase? I did most of the fixing by visual inspecting, but it would have been quite convenient if I could have somehow seen every time a function was called along with its arguments, kind of akin to a debugger in an imperative-style programming language.

Anyways, thanks!

**Comment by zliu41** [comment]
Fixed in https://github.com/ndmitchell/hlint/pull/1478
